### ISSUE TITLE
construction of URLs

### ISSUE THREAD

**Issue by julianhoersch** [COMCIFS/instrument-geometry-info#47]
I just create an issue from your mail @kroon-lab, to keep track
> If I give the URLs like they are on Zenodo, I can only give one name, whereas there are four different data files (_b4_1_000001.h5, _b4_1_000002._b4_1_000002.h5,h5,_b4_1_000002.h5). So giving one URL is not enough.

**Comment by julianhoersch** [comment]
I didn't consider multiple URLs yet. Should I just append the local filename to the given address or request multiple URLs containing the filename (this is maybe more difficult to map to the corresponding frames). I'm also not sure if the external URLs are created correctly for archives.

**Comment by jamesrhester** [comment]
I think @dallanto 's original software was able to handle multiple files and work out the number of frames, so that might have some useful bits in it. In terms of obtaining the file names, something like "To enter multiple files with a single base URL, finish the URL with a forward slash (`/`)".  Then if the user enters a URL with a final slash, prompt the user to enter filenames, with an empty line finishing the list.

**Comment by julianhoersch** [comment]
- Would it be legitimate to take the external links in the hdf5 master and use them for the corresponding frames? The program would then ask for the external url of the hdf5 master, look for the external links and construct the corresponding external url's from the address of the master file and the name of the external urls. 
- Is there a general need to be able to enter multiple url's also for cbf?

**Comment by jamesrhester** [comment]
Yes, I think looking into the hdf5 master to work out the external links is absolutely OK.  I have also seen datasets with cbf frames where they have been split over 4 .tgz files, so I think we need to cope with that.

**Comment by jamesrhester** [comment]
Here is an example of a dataset that has 4 distinct URLs: https://zenodo.org/record/2611942/
